### PR TITLE
add first batch of methods in oneflow.nn.functional namespace

### PR DESCRIPTION
**Pull request by doombeaker**
8.2日晚上布置的任务。
| functional API         | 认领人 | reviewer | PR   | 备注 | 放置到文件 |
| ---------------------- | ------ | -------- | ---- | ---- | ---------- |
| conv1d            |  张舸航   |          | #5693      | 默认参数不对齐     | conv.py |
| conv2d | 张舸航 | |#5693  |默认参数不对齐 | conv.py |
| conv3d | 张舸航 | |#5693  |默认参数不对齐 | conv.py |
| adaptive_avg_pool1d | 张舸航 | |#5693 | | pooling.py |
| adaptive_avg_pool2d | 张舸航 | | #5693| | pooling.py |
| adaptive_avg_pool3d | 张舸航 | | #5693| | pooling.py |
| relu | 张舸航 | |#5693 |  | activation.py |
| hardtanh | 张舸航 | |#5693 |默认参数不对齐 | activation.py |
| hardsigmoid | 张舸航 | |#5693 | | activation.py |
| hardswish | 张舸航 | |#5693 | | activation.py |
| leaky_relu | 张舸航 | |#5693 |默认参数不对齐 | activation.py |
| elu | 张舸航 | |#5693 |默认参数不对齐 | activation.py |
| selu | 张舸航 | |#5693 |默认参数不对齐 | activation.py |
| sigmoid | 张舸航 | |#5693 | | activation.py |
| prelu | 张婉茹 | |#5693 | | activation.py |
| gelu | 张婉茹 | |#5693 | | activation.py |
| logsigmoid | 张婉茹 | |#5693 | | activation.py |
| softsign | 张婉茹 | | #5693| | activation.py |
| softmax | 张婉茹 | |#5693 | | activation.py |
| softplus | 张婉茹 | |#5693 | | activation.py |
| tanh | 张婉茹 | |#5693 | | activation.py |
| silu | 张婉茹 | | #5693  | | activation.py |
| mish | 张婉茹 |  |#5693  | | activation.py |
| layer_norm | 张婉茹 | | |默认参数不对齐,二三参数含义不一样，不理解我们的二三参数的意义，还没问，没有写文档和测试样例 | | activation.py |
| dropout | 张婉茹 | | #5693 | | dropout.py |
| smooth_l1_loss | 张婉茹 | | |默认参数不对齐，不是很理解参数含义，没有写文档和测试样例 | loss.py |
| pad | 张婉茹 | |#5693 | 默认参数不对齐，我根据对ReplicationPad2d等类理解，自行修改的文档和测试样例，但是很可能有问题，待review | vision.py |
| upsample | 张婉茹 | |#5693  | 默认参数不对齐，我根据Upsample类的理解，自行修改的文档和测试样例，待review | vision.py |

- 默认参数不对齐情况
 torch：
image
image
image
image
image
image
image
image
image
image
image
image
image
image

oneflow：
image
image
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: python/oneflow/framework/docstr/pooling.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+adaptive_avg_pool2d(input, output_size) -> Tensor
+
+Applies a 2D adaptive average pooling over an input signal composed of
+    several input planes.
+
+See :class:`~oneflow.nn.AdaptiveAvgPool2d` for details and output shape.
+
+Args:
+    output_size: the target output size (single integer or
+        double-integer tuple)
```
Comment (doombeaker) [inline python/oneflow/framework/docstr/pooling.py:45]: 可以添加些 example，这个 pytorch 没有，不过我觉得我们添加 example 的话，这方面就比 pytorch 做得好

Comment (GehangZhang) [reply python/oneflow/framework/docstr/pooling.py:45]: ok，我是看torch源码没有

File: python/oneflow/framework/docstr/activation.py [hunk 2]
```diff
@@ -13,4 +13,17 @@
...
 limitations under the License.
 """
-from oneflow.nn.modules.interpolate import interpolate
+import oneflow
+from oneflow.framework.docstr.utils import add_docstr
+
+add_docstr(
+    oneflow.F.relu,
+    r"""
+    relu(Tensor x, *, Bool inplace=False) -> Tensor
```
Comment (doombeaker) [inline python/oneflow/framework/docstr/activation.py:22]: 我们的 relu 函数原型还不是这个吧？应该不支持 inplace 参数？

另外，Bool inplace 这种写法不是 python的语法（Bool也不是python类型），Python应该是：
`inplace: bool = False`。

其实不写注解也关系不大，语法上没错。写了当然更好。

File: docs/source/functional.rst [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction:: conv2d
+.. autofunction:: adaptive_avg_pool1d
+.. autofunction:: adaptive_avg_pool2d
+.. autofunction:: adaptive_avg_pool3d
+.. autofunction:: relu
+.. autofunction:: hardsigmoid
+.. autofunction:: hardswish
+.. autofunction:: sigmoid
+.. autofunction:: pad
+.. autofunction::prelu
```
Comment (doombeaker) [inline docs/source/functional.rst:16]:
```suggestion
.. autofunction:: prelu
```
少了空格，其它地方类似

File: docs/source/functional.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction::log_sigmoid
```
Comment (doombeaker) [inline docs/source/functional.rst:17]: 空格

File: docs/source/functional.rst [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction:: adaptive_avg_pool3d
+.. autofunction:: relu
+.. autofunction:: hardsigmoid
+.. autofunction:: hardswish
+.. autofunction:: sigmoid
+.. autofunction:: pad
+.. autofunction::prelu
+.. autofunction::log_sigmoid 
+.. autofunction:: gelu
+.. autofunction::softsign
```
Comment (doombeaker) [inline docs/source/functional.rst:19]: 空格

File: docs/source/functional.rst [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction::softmax
```
Comment (doombeaker) [inline docs/source/functional.rst:20]: 空格

File: docs/source/functional.rst [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction:: hardswish
+.. autofunction:: sigmoid
+.. autofunction:: pad
+.. autofunction::prelu
+.. autofunction::log_sigmoid 
+.. autofunction:: gelu
+.. autofunction::softsign
+.. autofunction::softmax 
+.. autofunction:: softplus
+.. autofunction::tanh
```
Comment (doombeaker) [inline docs/source/functional.rst:22]: 空格

File: docs/source/functional.rst [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction:: pad
+.. autofunction::prelu
+.. autofunction::log_sigmoid 
+.. autofunction:: gelu
+.. autofunction::softsign
+.. autofunction::softmax 
+.. autofunction:: softplus
+.. autofunction::tanh 
+.. autofunction:: silu
+.. autofunction::mish
```
Comment (doombeaker) [inline docs/source/functional.rst:24]: 空格

File: docs/source/functional.rst [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+.. autofunction::log_sigmoid 
+.. autofunction:: gelu
+.. autofunction::softsign
+.. autofunction::softmax 
+.. autofunction:: softplus
+.. autofunction::tanh 
+.. autofunction:: silu
+.. autofunction::mish 
+.. autofunction:: layer_norm
+.. autofunction::dropout
```
Comment (doombeaker) [inline docs/source/functional.rst:26]: 空格

File: docs/source/index.rst [hunk 10]
```diff
@@ -14,8 +14,10 @@ OneFlow API Reference
     oneflow
     tensor
     nn
+    functional
     module
     F
+    functional
```
Comment (doombeaker) [inline docs/source/index.rst:20]: 这个文件有一个 functional 就可以了。不要多次引用

File: python/oneflow/framework/docstr/activation.py [hunk 11]
```diff
@@ -0,0 +1,224 @@
...
+
+    See :class:`~oneflow.nn.Tanh` for more details.
+    
+    """,
+)
+
+add_docstr(
+    oneflow.F.silu,
+    r""" 
+    tanh(x: Tensor, *) -> Tensor
```
Comment (doombeaker) [inline python/oneflow/framework/docstr/activation.py:103]: 像这种 docstring，是不是都比较大段地参考了 pytorch，应该给出 pytorch 的文档的引用。
可以参考：
https://github.com/Oneflow-Inc/oneflow/blob/master/python/oneflow/nn/modules/normalization.py#L27

File: python/oneflow/framework/docstr/activation.py [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+
+See :class:`~oneflow.nn.Hardsigmoid` for more details.
+    """,
+)
+add_docstr(
+    oneflow.F.hardswish,
+    r"""
+hardswish(x: Tensor)-> Tensor
+
+Applies the hardswish function, element-wise, as described in the paper:
```
Comment (doombeaker) [inline python/oneflow/framework/docstr/activation.py:198]: 引用 pyotrch

File: oneflow/core/functional/functional_api.yaml [hunk 13]
```diff
@@ -724,7 +724,7 @@
 
 - name: "dropout"
   signature:
-    "Tensor Dropout(Tensor x, *, Float p, Generator generator=None)"
+    "Tensor Dropout(Tensor x, *, Float p=0.5, Generator generator=None)"
```
Comment (MARD1NO) [inline oneflow/core/functional/functional_api.yaml:727]: 可以在文档记录下，我们有个参数training没和torch对齐

File: python/oneflow/framework/docstr/activation.py [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+
+    Applies the rectified linear unit function element-wise. See
+    :class:`~oneflow.nn.ReLU` for more details.
+
+    """,
+)
+add_docstr(
+    oneflow.F.hardsigmoid,
+    r"""
+hardsigmoid(x: Tensor)-> Tensor
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/activation.py:289]: 缩进改一下

File: python/oneflow/framework/docstr/activation.py [hunk 15]
```diff
@@ -0,0 +1,424 @@
...
+Args:
+    inplace: If set to ``True``, will do this operation in-place. Default: ``False``
+
+See :class:`~oneflow.nn.Hardsigmoid` for more details.
+    """,
+)
+add_docstr(
+    oneflow.F.hardswish,
+    r"""
+hardswish(x: Tensor)-> Tensor
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/activation.py:309]: 缩进改下，后面不再赘述

File: python/oneflow/framework/docstr/dropout.py [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+       
+        >>> arr = np.array(
+        ...    [
+        ...        [-0.7797, 0.2264, 0.2458, 0.4163],
+        ...        [0.4299, 0.3626, -0.4892, 0.4141],
+        ...        [-1.4115, 1.2183, -0.5503, 0.6520],
+        ...    ]
+        ... )
+        >>> x = flow.Tensor(arr)
+        >>> y = flow.F.dropout(x,p=0)
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/dropout.py:57]: flow.F.dropout(x, p=0) 加个空格

File: python/oneflow/framework/docstr/math_ops.py [hunk 17]
```diff
@@ -18,9 +18,58 @@
...
+
+    .. code-block:: python
+
+        >>> import oneflow as flow
+        >>> import numpy as np
+        >>> x1 = flow.Tensor(np.array([-0.5461,  0.1347, -2.7266, -0.2746]).astype(np.float32))
+        >>> y1 = flow.F.sin(x1)
+        >>> y1
+        tensor([-0.5194,  0.1343, -0.4032, -0.2712], dtype=oneflow.float32)
+        >>> x2 = flow.Tensor(np.array([-1.4, 2.6, 3.7]).astype(np.float32),device=flow.device('cuda'))
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/math_ops.py:43]: device前面逗号加个空格，后不赘述 （当然确实是之前原文档的问题，但我觉得还是这里改好，不用后续留坑）

File: python/oneflow/framework/docstr/pooling.py [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+Args:
+    output_size: the target output size (single integer or
+        triple-integer tuple)
+
+For examples:
+
+.. code-block:: python
+
+    >>> import oneflow as flow
+    >>> import numpy as np
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/pooling.py:90]: import后换一行，后不赘述

File: python/oneflow/framework/docstr/activation.py (hunk 15 above)
Comment (1170500116) [reply python/oneflow/framework/docstr/activation.py:309]: 已改

File: python/oneflow/framework/docstr/math_ops.py (hunk 17 above)
Comment (1170500116) [reply python/oneflow/framework/docstr/math_ops.py:43]: 已改，只改了python/oneflow/framework/docstr下面的文件

File: python/oneflow/framework/docstr/dropout.py (hunk 16 above)
Comment (1170500116) [reply python/oneflow/framework/docstr/dropout.py:60]: 已改

File: python/oneflow/framework/docstr/activation.py (hunk 14 above)
Comment (1170500116) [reply python/oneflow/framework/docstr/activation.py:289]: 已改

File: oneflow/core/functional/functional_api.yaml (hunk 13 above)
Comment (1170500116) [reply oneflow/core/functional/functional_api.yaml:727]: 已改

File: python/oneflow/framework/docstr/activation.py [hunk 19]
```diff
@@ -0,0 +1,401 @@
...
+    .. math::
+        \text{log_sigmoid}(x) = \log\left(\frac{ 1 }{ 1 + \exp(-x)}\right)
+   
+    For example:
+
+    .. code-block:: python
+
+
+        >>> import numpy as np
+
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/activation.py:139]: 这里还是再麻烦改下吧，import可以不换行，import和下面的代码之间，换一行即可（后不赘述）

File: python/oneflow/framework/docstr/conv.py [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+    r"""
+    conv1d(input, weight, bias=None, stride=1, padding=0, dilation=1, groups=1) -> Tensor
+
+    Applies a 1D convolution over an input signal composed of several input
+    planes.
+
+
+    See :class:`~oneflow.nn.Conv1d` for details and output shape.
+
+        """,
```
Comment (MARD1NO) [inline python/oneflow/framework/docstr/conv.py:30]: 缩进对齐下（后不赘述

**Review by MARD1NO — APPROVED** [review]
